docs/api/definitionSetup.js.html

```diff
@@ -438,11 +429,36 @@ <h1 class="page-title">Source: jive-sdk-service/lib/definitionSetup.js</h1>
         };
 
         return remove( jive.tiles.definitions).then( function() {
-           return remove( jive.extstreams.definitions );
+            return remove( jive.extstreams.definitions );
         });
     });
 
-};</pre>
+};
+
+/**
+ * This will setup tiledefinition persistence by having one tile insert at a time.
+ * Tested on jive-persistence-postgres and jive-persistence-mongo
+ * @param app
+ * @param definitionsRootDir
+ * @param directoryContents
+ * @returns {*}
+ */
+function setupDefinitionsSequentially(app,definitionsRootDir, directoryContents) {
+    if (directoryContents.length == 0) {
+        return q.resolve();
+    } else {
+        var item = directoryContents.shift();
+        if (!isValidFile(item)) {
+            return setupDefinitionsSequentially(app,definitionsRootDir, directoryContents);
```

Comment:
By using q.all() we were asynchronously creating definitions in the database for all the tiles in the tiles directory. This works fine if we use mongodb,memory or file as persistence module. But when we try to use this for postgres as persistence module. 
It throws **error: duplicate key value violates unique constraint "pg_class_relname_nsp_index"** when we first time start the add-on service though it works fine afterwards.
On high level our understanding is that it is because of asynchronous nature of setupAllTileDefinition method. Because of which it is trying to create the index twice in parallel.
So we tried to change the nature setupAllTileDefinition to synchronous and it worked fine.